### PR TITLE
Update generate_rss.py

### PR DESCRIPTION
**Pull request by snootysteppes**
Key Improvements:
> Metadata: Added language, copyright, and dynamic last build date to the RSS feed metadata.
> Seen Comics Limiting: Capped the number of comics stored in seen_comics.json to 500 to keep the list manageable.
> Error Handling: Network requests now handle timeouts and other errors gracefully.
> Feed Description: Each comic's image is clickable and the alt text is displayed alongside the comic link in the feed description.
> Dynamic Build Date: The feed now updates with the current date and time whenever it's generated.

### CODE REVIEW

File: generate_rss.py [hunk 1]
```diff
@@ -13,55 +15,79 @@ def load_seen_comics():
             return json.load(file)
     return []  # If the file doesn't exist, return an empty list
 
-# Save seen comics to a file
+# Save seen comics to a file, limiting the size of the list
```
Comment (TzorAL) [inline generate_rss.py:18]: The purpose of this .json file is to ensure that the RSS feed does not display the same xkcd comic until every comic has been shown at least once. I intend to implement code that will reset the .json file once the number of viewed comics matches the total number of available comics.

File: generate_rss.py [hunk 2]
```diff
@@ -13,55 +15,79 @@ def load_seen_comics():
...
     save_seen_comics(seen_comics)
     
     title = comic_data['title']
     img_url = comic_data['img']
     alt_text = comic_data['alt']
+    pub_date = datetime.now().strftime("%a, %d %b %Y %H:%M:%S GMT")
 
     rss_content = f"""<?xml version="1.0" encoding="UTF-8" ?>
 <rss version="2.0">
   <channel>
```
Comment (TzorAL) [inline generate_rss.py:74]: The changes you've made to the rss feed are quite interesting. I'll check how it appears in the rss feed and see if they can be added to my repo😊

File: generate_rss.py [hunk 3]
```diff
@@ -13,55 +15,79 @@ def load_seen_comics():
...
+        seen_comics = seen_comics[-MAX_SEEN_COMICS:]  # Keep only the last MAX_SEEN_COMICS entries
+    
     with open(SEEN_COMICS_FILE, 'w') as file:
         json.dump(seen_comics, file)
 
 # Fetch the latest XKCD comic number
 def fetch_latest_comic_number():
-    response = requests.get('https://xkcd.com/info.0.json')
-    latest_comic_data = response.json()
-    return latest_comic_data['num']
```
Comment (TzorAL) [inline generate_rss.py:25]: Yeah, it's propably a good idea for me to add error catch blocks as well😅

File: generate_rss.py (hunk 1 above)
Comment (snootysteppes) [reply generate_rss.py:18]: Makes more sense, thanks.

**Comment by snootysteppes** [comment]
Hey there, is this OK enough to be merged, or would you like more changes sir?

**Pull request closed without merging**